### ISSUE TITLE
Should complain about unquoted strings in array assignments

### ISSUE THREAD

**Issue by LukeShu** [koalaman/shellcheck#864]
#### For bugs
- Rule Id (if any, e.g. SC1000):  SC2086
- My shellcheck version (`shellcheck --version` or "online"): 0.4.6
- [X] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### For new checks and feature suggestions
- [X] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [ ] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

This may be related to #790 .

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash
s='*'
a=($s)
printf '%s\n' "${a[@]}"
```

#### Here's what shellcheck currently says:

    No issues detected!

#### Here's what I wanted or expected to see:

```
Line 3:
a=($s)
   ^-- SC2086: Double quote to prevent globbing and word splitting.
```

Since in such an assignment, globbing and word splitting may happen, I'd expect shellcheck to invoke SC2086.  I suspect that what is happening is that it is treating this assignment like a non-array assignment (eg, `foo=$bar`), in which the globbing and word splitting wouldn't happen.

**Comment by koalaman** [comment]
The root cause is actually intentional; it doesn't trigger for the same reason `for var in $string` doesn't.

I agree though. There should be a warning, even though this is a common way to intentionally split things.  Maybe a separate message that's more helpful in the array case.

**Comment by koalaman** [comment]
This has been fixed in 3c75674. Two new warnings have been added:
